packages/iris-grid/src/IrisGrid.tsx

```diff
@@ -2453,6 +2453,29 @@ export class IrisGrid extends Component<IrisGridProps, IrisGridState> {
       this.focusRowInGrid(row);
       return;
     }
+
+    let cursorRow;
+    let cursorColumn;
+    if (this.grid) {
+      ({ cursorRow, cursorColumn } = this.grid.state);
+    }
+    if (cursorRow != null && cursorColumn != null) {
+      // if a row is selected
```

Comment:
Flip the logic here to do an early return if nothing is selected. That way the majority of the logic is at the lowest possible indent level

```typescript
if (cursorRow == null || cursorColumn == null) {
  this.setState({
    // All the empty strings from the end of this function
  });
}

// Do the value fetch and set state
```